Session.cc

```diff
@@ -1279,6 +1284,8 @@ bool Session::SendSpectralProfileData(int file_id, int region_id, bool stokes_ch
     if ((region_id > CURSOR_REGION_ID) || (region_id == ALL_REGIONS) || (file_id == ALL_FILES)) {
         // Region spectral profile
         CARTA::SpectralProfileData profile_data;
+        _moment_controller->StopCalculation(file_id); // Stop the moment calculations while spectral profile begins
```

Comment:
As shown in the if statement, file_id could be ALL_FILES (-1) which would not be in the moment controller's map.  It is ALL_FILES when the region changes so that the spectral profiles for all images with that region are recalculated.  In RegionHandler::FillSpectralProfileData, it iterates through the spectral configs (_spectral_req) to start a new profile when the requirement contains the region_id.  This is why I had recommended that the RegionHandler handle the moments for regions, as it could call StopCalculation for each actual file_id in this loop before doing the spectral profile.  The simplest thing to do for ALL_FILES at this point (before the refactor) is to add a RegionHandler function to return the file ids for a given region id by checking the _spectral_req ConfigId keys.  ConfigId is a struct containing file_id and region_id.  Also note that the key may exist with no SpectralConfigs assigned; the key is removed when the region or frame is closed.

Reply:
Thank you for the suggestion! Have modified the code accordingly. Please check is that ok.